src/Silk.Core.Discord/Main.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using DSharpPlus;
+using DSharpPlus.CommandsNext;
+using DSharpPlus.Interactivity.Extensions;
+using MediatR;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Silk.Core.Discord.EventHandlers;
+using Silk.Core.Discord.EventHandlers.MemberAdded;
+using Silk.Core.Discord.EventHandlers.Notifications;
+using Silk.Core.Discord.Types;
+using Silk.Core.Discord.Utilities;
+using Silk.Core.Discord.Utilities.Bot;
+using Silk.Core.Discord.Utilities.HelpFormatter;
+using Silk.Extensions;
+
+namespace Silk.Core.Discord
+{
+    public class Main : IHostedService
+    {
+        public static BotState State { get; private set; } = BotState.Starting;
+        //public static DiscordSlashClient SlashClient { get; } // Soon™ //
+        public static DiscordShardedClient ShardClient { get; private set; }
```

Comment:
Why is this static in a hosted service? Nothing should be able to grab your bot service until it's initialized. I personally suggest breaking your bot service and main service into two separate classes. Make your bot service not a hosted service but a Singleton and then have your Main IHostedService consume it. That's the way I personally did it but honestly, this may work just as well.

Reply:
Meant to change these to be instanced.